plugins/stamina/sh_plugin.lua

```diff
@@ -80,6 +67,11 @@ local function CalcStaminaChange(client)
 	end
 end
 
+function PLUGIN:SetupMove(client, cMoveData)
```

Comment:
Setting max speed in... SetupMove?

Reply:
No you are totally right to do this, it's the best way to control this independently from `Player:SetWalk(Run)Speed`. Calling `CMoveData:SetMaxClientSpeed` in a [SetupMove](https://wiki.facepunch.com/gmod/GM:SetupMove) hook also calls `CMoveData:SetMaxSpeed`. The implementation is fine.

Reply:
Yeah I have also realized that this is the case-- I haven't done non-nextbot loco in awhile.